stl/inc/flat_set

```diff
@@ -180,25 +180,27 @@ public:
     }
 
     _Deriv& operator=(initializer_list<_Kty> _Ilist) {
-        _Clear_guard<_Base_flat_set> _Guard{this};
+        _Clear_guard<container_type> _Guard{_STD addressof(_Mycont)};
         _Mycont.assign(_Ilist);
         _Make_invariants_fulfilled();
         _Guard._Target = nullptr;
         return static_cast<_Deriv&>(*this);
     }
 
     // iterators
+    // for flat_meow, `const_iterator` is required to be convertible to `iterator` (per N4958
+    // [associative.reqmts.general]/6)
     _NODISCARD iterator begin() noexcept {

```

Comment:
Why not remove this overload, leaving only `begin() const`? (Ditto for `end()` on line 199.)